scarb/tests/e2e/add.rs

```diff
@@ -350,3 +350,61 @@ fn overwrite_change_source_from_path_to_git() {
         "#})
         .run();
 }
+
+#[test]
+fn should_sort_if_already_sorted() {
+    ManifestEditHarness::offline()
+        .args(["add", "cat@2.0.0"])
+        .input(indoc! {r#"
+            [package]
+            name = "hello"
+            version = "1.0.0"
+
+            [dependencies]
+            bar = "1.0.0"
+            dep = "1.0.0"
+            foo = "1.0.0"
```

Comment:
how would this behave in such scenario, when adding `cat`?

```diff
  [dependencies]
  bar = "1.0.0"
+
  dep = "1.0.0"
  foo = "1.0.0"
```

I'm interested on what happens with the blank line

Reply:
Ah.. this would just append the entry before the blank line, if it's already sorted. You reckon I should put that in the tests as well?

Reply:
Yeah, please definitely add a test, so that we can keep an eye on this behaviour. No need to deal with it explicitly, it's fine what it does!